libcst/codemod/visitors/_imports.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from dataclasses import dataclass, replace
+from typing import Optional
+
+from libcst.helpers import get_absolute_module
+
+
+@dataclass(frozen=True)
+class ImportItem:
+    """Representation of individual import items for codemods."""
+
+    module_name: str
+    obj: Optional[str] = None
+    alias: Optional[str] = None
+    relative: int = 0
+
+    def __post_init__(self) -> None:
+        if self.module_name is None:
+            object.__setattr__(self, "module_name", "")
+        elif self.module_name.startswith("."):
+            mod = self.module_name.lstrip(".")
+            rel = self.relative + len(self.module_name) - len(mod)
+            object.__setattr__(self, "module_name", mod)
+            object.__setattr__(self, "relative", rel)
+
+    @property
+    def module(self) -> str:
+        return "." * self.relative + self.module_name
+
+    def resolve_relative(self, base_module: Optional[str]) -> "ImportItem":
+        """Return an ImportItem with an absolute module name if possible."""
+        mod = self
+        # `import ..a` -> `from .. import a`
+        if mod.relative and mod.obj is None:
+            # py3.6 dataclass backport does not like `obj` as an arg to replace()
+            mod = ImportItem(
+                module_name="",
+                obj=mod.module_name,
+                alias=mod.alias,
+                relative=mod.relative,
+            )
```

Comment:
I don't really like this because we'll have to remember to update this call when we add/remove fields to/from `ImportItem`. It'd be easier to just rename the `obj` field to `obj_name`

Reply:
yeah, good point

Reply:
done